dev-docs/RFCs/proposals/transport-format-rfc.md

```diff
@@ -0,0 +1,110 @@
+# RFC: Transport Format RFC
+
+* Authors: Xiaoji Chen
+* Date: Jan, 2020
+* Status: **Draft**
+
+## Summary
+
+This RFC proposes a generic transport format for generating/rendering data cross platforms.
+
+## Background
+
+In v7.3, deck.gl added the `@deck.gl/json` module. It supports generating serializable deck.gl props in non-JavaScript environments and send it to render on the client side. This format is currently used by `pydeck`, deck.gl's Python binding.
+
+We have plans to add bindings for other languages, and potentially non-JS renderers. This calls for a more comprehensive format that can be easily generated, sent, and handled by multiple languages and platforms.
+
+The JSON format has the following shortcomings:
+
+- Efficiency: Serializing/deserializing large data objects with JSON is expensive in terms of both time and space. Network bandwidth, CPU time and memory are all valuable assets when it comes to web applications.
+- Updates: It is not possible to update a layer without re-sending the entire snapshot. This is important for animation and/or live data visualization.
+
+A hybrid transport format of both JSON and binary will help address these issues.
+
+## Proposal
+
+Expand the `JSONConverter.convert` API to accept a second argument:
+
+```js
+converter.convert(json, buffers);
+```
+
+Where `buffers` is a list/map of ArrayBuffers.
+
+In `json`, binary data can be referenced using the following syntax:
+
+```
+@@binary[<id>]/<type>/<offset>/<length>
```

Comment:
Nit: Is there a strong reason to have a separate text syntax here for the type, instead of reusing the accessor props from the JS API? Thinking about the high-level "One API" principle...

The `GL` constants should be part of the configuration.

Reply:
I agree. I will update the proposal.